src/app/loan-form/loan-form.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { Subject } from 'rxjs';
+import { FloatLabelType } from '@angular/material/form-field';
+
+@Component({
+  selector: 'app-loan-form',
+  templateUrl: './loan-form.component.html',
+  styleUrls: ['./loan-form.component.scss'],
+})
+export class LoanFormComponent implements OnInit, OnDestroy {
+  private destroy$ = new Subject<void>();
+  loanForm: FormGroup;
+  constructor(private fb: FormBuilder) {
+    this.loanForm = this.fb.group({
+      totalAmount: ['', [Validators.required, Validators.min(10000), Validators.pattern("^[0-9]*$")]],
+      downPayment: ['', [Validators.required]],
+      termYears: [
+        '',
+        [Validators.required, Validators.min(1), Validators.max(30), Validators.pattern("^[0-9]*$")],
+      ],
+    });
+
+    this.loanForm.get('totalAmount')!.valueChanges.subscribe((totalAmount) => {
+      const downPaymentControl = this.loanForm.get('downPayment');
+      if (totalAmount >= 1000) {
+        if (!downPaymentControl?.value || downPaymentControl?.pristine) {
+          const downPayment = totalAmount * 0.15;
+          downPaymentControl?.setValue(downPayment);
+        }
+        downPaymentControl?.setValidators([
```

Comment:
You can use cross field validators for this, so you don't need to do this on every value change. 
https://angular.io/guide/form-validation#adding-cross-validation-to-reactive-forms